app/javascript/react/screens/App/Plan/PlanReducer.js

```diff
@@ -50,21 +56,32 @@ const excludeDownloadDoneTaskId = (allDownloadLogInProgressTaskIds, taskId) =>
 const includeDownloadInProgressTaskId = (allDownloadLogInProgressTaskIds, taskId) =>
   allDownloadLogInProgressTaskIds ? allDownloadLogInProgressTaskIds.concat(taskId) : [taskId];
 
+const removeCancelledTasksFromSelection = (allSelectedTasksForCancel, alreadyCancelledTasks) => {
```

Comment:
This is fine, but a little odd looking, and will run in n^2 time.  Here's my approach on this one, taking the map out and only using one `filter`, and then using `every` which will short circuit if it sees a non-matching element.

Not tested, but:
```
const removeCancelledTasksFromSelection = (allSelectedTasksForCancel, alreadyCancelledTasks) => (
  allSelectedTasksForCancel.filter(selectedTask =>
    alreadyCancelledTasks.every(cancelledTask => selectedTask.id !== cancelledTask.id)
  )
);
```

Not that we need to optimize for runtime here, but I think that's more readable too.

Reply:
Good suggestion.

I have applied this change in the last commit.